beetsplug/tidal.py

```diff
@@ -0,0 +1,1050 @@
+            return None
+
+        # Return either synced lyrics or unsynced depending on config and availability
+        if self.config["synced_lyrics"]:
+            if lyrics.subtitles:
+                self._log.debug(
+                    f"Synced lyrics are available for track {track.id}"
+                )
+                return lyrics.subtitles
+            else:
+                self._log.info(
+                    (
+                        f"Synced lyrics not available for track {track.id},"
+                        "returning unsynced lyrics"
+                    )
+                )
+                return lyrics.text
+        else:
+            return lyrics.text
+
+    def _validate_lyrics(self, lib_item, tidal_item):
+        """Validates lyrics retrieved from TIDAL
+
+        Currently we just use the difference of length in the TIDAL Item vs
+        the Library item.
+
+        :param lib_item: Item to validate from the Library
+        :type lib_item: beets.library.Item
+        :param tidal_item: Item to validate from TIDAL
+        :type tidal_item: tidalapi.media.Track
+        :return: If the lyrics are valid or not
+        :rtype: bool
+        """
+        self._log.debug(f"Validating lyrics for {lib_item.title}!")
+        maxdiff = self.config["max_lyrics_time_difference"].as_number()
+
+        if maxdiff >= 1:
+            # Validate that both the item and the Tidal metadata have values
+            if not lib_item.length or tidal_item.duration == -1:
+                self._log.debug(
+                    (
+                        "Not using duration difference to validate lyrics "
+                        "as one or both items don't have a duration!"
+                    )
+                )
+                return bool(self.config["lyrics_no_duration_valid"])
+
+            self._log.debug("Using duration difference to validate lyrics")
+            self._log.debug(
+                (
+                    f"Item duration: {lib_item.length}, "
+                    f"Tidal duration: {tidal_item.duration}"
+                )
+            )
+
+            # Calculate difference
+            difference = abs(lib_item.length - tidal_item.duration)
+            difference_over = abs(maxdiff - difference)
+
+            if difference > maxdiff:
+                self._log.debug(
+                    (
+                        f"Not using lyrics for {lib_item.title}"
+                        f"as difference is {difference_over} over the max of {maxdiff}"
+                    )
+                )
+                return False
+        else:
+            self._log.debug(
+                (
+                    "Not using timestamp lyrics validation "
+                    "due to user configuration"
+                )
+            )
+
+        # Nothing above invalidated the lyrics, assuming valid
+        return True
+
+    def _search_lyrics(self, item, limit=10):
```

Comment:
I don't think this is the right place for lyrics search. We have `lyrics` plugin for this, so I would expect TIDAL to be added as a backend there (in a separate PR).

Reply:
That's what I thought at first, but I though that from https://github.com/beetbox/beets/pull/4641#issuecomment-1562128419 that they wanted it to be part of the autotagger.

Anyways, would I be reimplementing parts of this plugin in the lyrics plugin or would I be able to call into this plugin from the lyrics plugin? Either that or monkeypatch the lyrics plugin to add a new backend